src/app/components/Shoutboard.tsx

```diff
@@ -26,78 +24,93 @@ import TextField from '@material-ui/core/TextField';
         super(props)
-        this._onClickHandler = this._onClickHandler.bind(this)
     }
 
-    @observable counter: number = 0
+  postsStore = new PostsStore()
+  @observable counter: number = 0
+
+  @observable isVisible = false
+
+  @action.bound private toggleComponent() {
+    //console.log(this.visible)
+    this.isVisible = !this.isVisible
+  }
 
 
-    _onClickHandler() {
-        if(!routeState.visible){
-            //console.log(true)
-            routeState.showComponent(true)
-        }
-        else{
-            //console.log(false)
-            routeState.showComponent(false)
-        }
-    }
 
     render() {
         return (
-        	<div className="h-screen flex flex-col">
-                <div className="flex-none">
-	                <Nav />
-                </div>
-                <div className="flex-1 overflow-y-auto p-4">
-	                <h1 className="text-center">Shoutboard will be here</h1>
-                    <Button className="bg-grey-light" onClick={() => this._onClickHandler()}> Create Post </Button>
-                    <hr/>
-                    {routeState.visible && <CreatePost/>}
-                    {routeState.posts.map((item) => {
-                        return <div className="container pb-2 lg:flex" key={item._id}>
-                            <div className="w-full border border-grey-light lg:border-b lg:border-t lg:border-grey-light bg-white rounded-b lg:rounded-b lg:rounded-r p-4 flex flex-col justify-between leading-normal">
-                                <h2>{item.name}</h2>
-                                <p>{item.text}</p>
-                            </div>
-                        </div>
-                    })}
+          <Provider postsStore={this.postsStore}>
+          	<div className="h-screen flex flex-col">
+                  <div className="flex-none">
+  	                <Nav />
+                  </div>
+                  <div className="flex-1 overflow-y-auto p-4">
+  	                <h1 className="text-center">Shoutboard will be here</h1>
+                      <Button className="bg-grey-light" onClick={this.toggleComponent}> Create Post </Button>
+                      <hr/>
+                      {this.isVisible && <CreatePost/>}
+                      {this.postsStore.posts.map((item) => {
+                          return(
+                          <div className="container max-w-full pb-2 lg:flex" key={item.id}>
+                              <div className="w-full border border-grey-light lg:border-b lg:border-t lg:border-grey-light rounded lg:rounded p-4 shadow-md rounded">
+                                  <Typography variant="title">{item.name}</Typography>
+                                  <Typography>{item.text}</Typography>
+                              </div>
+                          </div>
+                          )
+                      })}
                 </div>
             </div>
+          </Provider>
         )
     }
 }
 
+
+
+class Post {
```

Comment:
I think this can already be moved to separate file, so you can import it in PostsStore. Otherwise there would be a import cycle (PostsStore imports Shoutboard as needs Post, Shoutboard imports PostsStore as it needs store)